opensrp-chw/src/main/java/org/smartregister/chw/activity/ReferralFollowupActivity.java

```diff
@@ -96,6 +96,15 @@ protected void completeReferralTask(String jsonString) {
                     // update task
                     TaskRepository taskRepository = ChwApplication.getInstance().getTaskRepository();
                     Task task = taskRepository.getTaskByIdentifier(getTaskIdentifier());
+
+                    //Get task status before updating
+                    Task.TaskStatus status = task.getStatus();
+                    if (status.equals(Task.TaskStatus.IN_PROGRESS)){ //Task was attended at the facility/Addo
+                        task.setBusinessStatus(org.smartregister.chw.util.Constants.TaskBusinessStatus.ATTENDED);
+                    }else if (status.equals(Task.TaskStatus.READY)){ //Task was not attended at the facility/Addo
+                        task.setBusinessStatus(org.smartregister.chw.util.Constants.TaskBusinessStatus.UNATTENDED);
```

Comment:
Hi @issyzac Does this mean the status can be `IN_PROGRESS` with `TaskBusinessStatus.UNATTENDED` or even `READY` with `TaskBusinessStatus.ATTENDED` 🤔

Reply:
Also probably the implementation that makes sense in all this is to update the business status at the ADDO/HF level on an instance where the tasks are being acted upon which will surely make so much sense but also this is an attempt to try and solve the visibility issue without having to issue an update across the board in all the Apps 😆